File: CSE 241 - Object Oriented Programming/Homework 5/Homework/BoardArray1D.cpp
#include "BoardArray1D.h"

BoardArray1D::~BoardArray1D(){
	delete[] board;
}

void BoardArray1D::setCellValue(int y, int x, int value){
	board[y * boardSizey + x] = value;
}

void BoardArray1D::setSize(int sizeX, int sizeY){
	delete[] board;
	
	boardSizex = sizeX;
	boardSizey = sizeY;
	
	board = new int[sizeY * sizeX];
}

int BoardArray1D::operator () (int y, int x){
	if(y < boardSizey && x < boardSizex){
		return board[y * boardSizey + x];
  } else{
		exit(0);
  }
}

File: CSE 241 - Object Oriented Programming/Homework 7/Homework/BoardArray1D.java
/**
 * A board class that represents the board as 1D array
 * @author Berke Belgin
 */
public class BoardArray1D extends AbstractBoard{
	private int[] board;
	
	public BoardArray1D(){
		super();
	}
	public BoardArray1D(int y, int x){
		super(y, x);
	}
	
	protected void finalize(){
		super.finalize();
	}
	
	public void setCell(int y, int x, int value){
		board[y * sizeY + x] = value;
	}
	public void setSize(int sizeY, int sizeX){
		board = new int[sizeY * sizeX];
		
		this.sizeY = sizeY;
		this.sizeX = sizeX;
	}
	public int cell(int y, int x){
		if(y < sizeY && x < sizeX){
			return board[y * sizeY + x];
		} else{
			System.exit(0);
			return 0;
		}
	}
}
